jetty-security/src/main/java/org/eclipse/jetty/security/SecurityHandler.java

```diff
@@ -76,11 +77,8 @@
 
     static
     {
-        for (Authenticator.Factory factory : ServiceLoader.load(Authenticator.Factory.class))
-        {
-            __knownAuthenticatorFactories.add(factory);
-        }
-
+        ServiceLoader.load(Authenticator.Factory.class).stream()
+            .flatMap(TypeUtil::providerMap).forEach(__knownAuthenticatorFactories::add);
```

Comment:
This can be a collect

Reply:
We also need to add `new DefaultAuthenticatorFactory()` but `Collectors.toList()` may return an immutable list.